File: Easy/0682_baseball-game/Solution.cpp
#include <vector>
#include <string>

class Solution {
public:
    int calPoints(std::vector<std::string>& operations) {
        std::vector<int> stack;
        for (const std::string& op : operations) {
            if (op == "C") {
                stack.pop_back();
            } else if (op == "D") {
                stack.push_back(2 * stack.back());
            } else if (op == "+") {
                int n = stack.size();
                stack.push_back(stack[n - 1] + stack[n - 2]);
            } else {
                stack.push_back(std::stoi(op));
            }
        }
        int sum = 0;
        for (int score : stack) {
            sum += score;
        }
        return sum;
    }
};

File: Easy/0682_baseball-game/Solution.ts
function calPoints(operations: string[]): number {
    const stack: number[] = [];
    for (const op of operations) {
        if (op === "C") {
            stack.pop();
        } else if (op === "D") {
            stack.push(2 * stack[stack.length - 1]);
        } else if (op === "+") {
            stack.push(stack[stack.length - 1] + stack[stack.length - 2]);
        } else {
            stack.push(parseInt(op, 10));
        }
    }
    return stack.reduce((sum, score) => sum + score, 0);
}